lib/room_screen.dart

```diff
@@ -0,0 +1,54 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:flutter_webrtc/webrtc.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+
+import 'cable_provider.dart';
+import 'membership_pair_entry.dart';
+import 'room_channel.dart';
+
+class RoomScreen extends HookWidget {
+  final RTCVideoRenderer localRenderer;
+  final String id;
+
+  const RoomScreen({ Key key, @required this.localRenderer, @required this.id }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    final cable = useProvider(cableProvider);
+    final membershipPairEntries = useState<List<MembershipPairEntry>>([]);
+    final roomChannel = useState(
+      RoomChannel(
+        cable: cable,
+        id: id,
+        onMembershipPairEntryCreated: ({ membershipPairId, older }) {
+          membershipPairEntries.value = [
+            ...(membershipPairEntries.value),
+            MembershipPairEntry(membershipPairId: membershipPairId, older: older)
+          ];
```

Comment:
this feels good to me.

we have some meaningful state which is just a list of these pair entries, which we mutate in response to cable events, separate from the state of the cable itself

we're missing a uniqueness check on the membership pair ID though